data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticle.java

```diff
@@ -220,26 +228,32 @@ public void handle(HttpServerRequest request) {
     }
 
     public void updateServerConfig() {
+
         // This function will be called when the secret volume is updated
         File tlsKeyFile = new File(tlsKeyFilePath);
         File tlsCrtFile = new File(tlsCrtFilePath);
 
         // Check whether the tls.key and tls.crt files exist
         if (tlsKeyFile.exists() && tlsCrtFile.exists() && httpsServerOptions != null) {
-
-            // Update SSL configuration by using updateSSLOptions
-            PemKeyCertOptions keyCertOptions =
-                    new PemKeyCertOptions().setKeyPath(tlsKeyFile.getPath()).setCertPath(tlsCrtFile.getPath());
-
-            // result is a Future object
-            Future<Void> result = httpsServer.updateSSLOptions(new SSLOptions().setKeyCertOptions(keyCertOptions));
-
-            result.onSuccess(v -> {
-                        logger.info("Succeeded to update TLS key pair");
-                    })
-                    .onFailure(e -> {
-                        logger.error("Failed to update TLS key pair", e);
-                    });
+            try {
+                // Update SSL configuration by using updateSSLOptions
+                PemKeyCertOptions keyCertOptions = new PemKeyCertOptions()
+                        .setCertValue(Buffer.buffer(java.nio.file.Files.readString(tlsCrtFile.toPath())))
+                        .setKeyValue(Buffer.buffer(java.nio.file.Files.readString(tlsKeyFile.toPath())));
+
+                // result is a Future object
+                Future<Void> result = httpsServer.updateSSLOptions(new SSLOptions().setKeyCertOptions(keyCertOptions));
+
+                result.onSuccess(v -> {
+                            logger.info("Succeeded to update TLS key pair");
+                        })
+                        .onFailure(e -> {
+                            logger.error("Failed to update TLS key pair while executing updateSSLOptions", e);
+                        });
```

Comment:
Can we just chain `onSuccess` and `onFailure` to `updateSSLOptions` without intermediate variables?

Also lambdas with only a single statement can be collapsed and we can avoid using `v -> { <single statement> }` becomes `v -> <single statement>`, it's easier to read